packages/controllers/src/admin.rs

```diff
@@ -32,17 +34,17 @@ impl<'a> Admin<'a> {
         Admin(Item::new(namespace))
     }
 
-    pub fn set(&self, deps: DepsMut, admin: Option<Addr>) -> StdResult<()> {
+    pub fn set<Q: CustomQuery>(&self, deps: DepsMut<Q>, admin: Option<Addr>) -> StdResult<()> {
```

Comment:
Why not defining `Q` over the struct? That way it can have a default of `Empty`.

You would require a `PhantomData`, but it's probably worth it.

The same with the other structs.

Reply:
It seems doing that is more work than I anticipated... your call.

Reply:
I'll take a look!

Reply:
I wonder about this. From another point of view, I feel like `Admin` shouldn't be concerned with what the custom query type is.

And I'm not sure the default of `Empty` is such a big benefit - in typical calls of these functions type inference should do the trick without the need to write something like `set::<TgradeQuery>(...)`... I think?

Reply:
For example, this line still works just fine after adding the type parameter to `set`:

https://github.com/CosmWasm/cw-plus/blob/controllers-custom-query-type/contracts/cw20-ics20/src/contract.rs#L42

Reply:
Yeah, I'm keeping this as is.